src/maths_utils.c

```diff
@@ -74,3 +74,21 @@ uint8_t ulog2(uint32_t value)
 	return (sizeof(uint8_t) * 8U) - result;
 #endif
 }
+
+uint32_t swd_odd_parity(const uint32_t value)
+{
+#if defined(__GNUC__)
+	/* Ask for the libgcc impl */
+	return __builtin_parity(value);
+#elif defined(_MSC_VER)
+	/* Ask for a CPU insn */
+	return __popcount(value) & 1U;
+#else
+	/* Generic impl */
+	uint32_t result = 0;
```

Comment:
With the change below, this can become a `uint8_t` or a `bool` - your choice - and the result type changed to match.

Reply:
Most hosted code uses `uint8_t` for packets represented as arrays. I'm more inclined to swap the `bool` in common/swdptap.c to uint8_t, or even `uint32_t`, which is faster (no bitmasking in codegen).
The original libgcc2 `__builtin_parity()` and `__builtin_popcount()` take `unsigned int` and return `int`, which is why I picked `uint32_t` to be transparent enough in the shim case. MSVC docs say `unsigned int __popcnt(unsigned int value);`, supporting this. In the "generic impl" case (on exotic toolchains?) I think we still should aim to return the same "int". Of course, if BMD only requires a 1 or a 0 from it, then it can be demoted to `uint8_t` everywhere, unless the hosted code already does this.

Reply:
The function can only produce the values '0' and '1', so hence suggesting using `uint8_t` or `bool` - given that point about uint8_t for packets and the fact this get used to initialise values in those packets, it supports going to `uint8_t` and doing all the type truncation here - this will eliminate a pile of `-Wconversion`/`-Warith-conversion` warnings and we don't think will negatively impact codegen.

Reply:
Adjusted to `uint8_t`.